src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -2002,7 +2006,7 @@ void InterpreterCreateQuery::checkDatabaseNameAllowed() {
     if (internal)
         return;
     auto *storage = create.storage;
-    if (!storage || !storage->engine || storage->engine->name != "Replicated")
+    if (storage && storage->engine && storage->engine->name != "Replicated")
```

Comment:
can this change be harmful. 
earlier withtout `storage` or without `storage->engine` the function would return. and now it wont.

Reply:
I guess it is fine.
Because this function needs to check only replicated database names and its behavior was changed.
Before replicated database could have been created only by explicitly specifying engine `Replicated`, so, if there were no storage -> it is not replicated.

But now, absence of storage -> engine = `Replicated`, which means we have to validate the name in this case too.